File: Golang/SearchForARange/SearchForARange_test.go
package SearchForARange

import (
	"testing"
	"reflect"
)

func TestSearchRange(t *testing.T) {

	if reflect.DeepEqual(searchRange([]int{1}, 1),
		[]int{0, 0}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(searchRange([]int{5, 7, 7, 8, 8, 10}, 11),
		[]int{-1, -1}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	if reflect.DeepEqual(searchRange([]int{5, 7, 7, 8, 8, 10}, 8),
		[]int{3, 4}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/SearchForARangeTest.py
from unittest import TestCase

from SearchForARange import SearchForARange


class TestSearchForARange(TestCase):
    def test_searchRange(self):
        sr = SearchForARange()

        self.assertEqual(sr.searchRange([1], 1), [0, 0])

        self.assertEqual(sr.searchRange([5, 7, 7, 8, 8, 10], 11), [-1, -1])

        self.assertEqual(sr.searchRange([5, 7, 7, 8, 8, 10], 8), [3, 4])
